src/UBAFeeCalculator/UBAFeeConfig.ts

```diff
@@ -29,14 +30,32 @@ class UBAConfig {
    */
   private readonly balancingFee: DefaultOverrideStructure<TupleParameter[], ChainId>;
 
+  /**
+   * A record of boundry values for each chain and token that define the threshold for when the
+   * running balance should be balanced back to a fixed amount. Due to the fact that this type of
+   * operation is based on a heuristic and is considered a non-event transient property of the
+   * protocol, the threshold must be computed based on the current running balance of the spoke from
+   * the last validated running balance.
+   */
+  private readonly balanceTriggerThreshold: Record<ChainTokenCombination, TupleParameter>;
+
+  /**
+   * Instantiate a new UBA Config object
+   * @param baselineFee A baseline fee that is applied to all transactions to allow LPs to earn a fee
+   * @param utilizationFee A record of piecewise functions for each chain that define the utilization fee to ensure that the bridge responds to periods of high utilization
+   * @param balancingFee A record of piecewise functions for each chain and token that define the balancing fee to ensure either a positive or negative penalty to bridging a token to a chain that is either under or over utilized
+   * @param balanceTriggerThreshold A record of boundry values for each chain and token that define the threshold for when the running balance should be balanced back to a fixed amount. Due to the fact that this type of operation is based on a heuristic and is considered a non-event transient property of the protocol, the threshold must be computed based on the current running balance of the spoke from the last validated running balance.
+   */
   constructor(
     baselineFee: DefaultOverrideStructure<BigNumber, RouteCombination>,
     utilizationFee: BigNumber,
-    balancingFee: DefaultOverrideStructure<TupleParameter[], ChainId>
+    balancingFee: DefaultOverrideStructure<TupleParameter[], ChainId>,
+    balanceTriggerThreshold: Record<ChainTokenCombination, TupleParameter>
```

Comment:
Just to make sure I understand, this trigger threshold is defined in the mainnet ConfigStore contract, right?

Do you think it would be better to just query the ConfigStoreClient directly instead of trying to track that state here?

Reply:
I've tried to leave this code in a state where it's as abstracted away from RPC calls as possible. Since this is purely a library wrapper over mathematical operations, I wanted to leave the RPC calls explicitly to the user/calling code.